docker/docker_client.go

```diff
@@ -342,7 +338,7 @@ func SearchRegistry(ctx context.Context, sys *types.SystemContext, registry, ima
 		hostname = dockerV1Hostname
 	}
 
-	client, err := newDockerClient(sys, hostname, registry)
+	client, err := newDockerClient(sys, hostname)
```

Comment:
Passing `dockerV1Hostname` (only) would change behavior, at least for `certDir`.

Reply:
I switched back to the initial master state, so the issue should be gone now.